### PR TITLE
[DOC] Enhanced RDoc for String #tr and #tr!

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: string.c [hunk 1]
```diff
@@ -8022,37 +8022,50 @@ rb_str_tr_bang(VALUE str, VALUE src, VALUE repl)
...
- *     "hello".tr('aeiou', '*')    #=> "h*ll*"
- *     "hello".tr('aeiou', 'AA*')  #=> "hAll*"
+ *  - A circumflex (<tt>'^'</tt>) in +replaceables+
+ *    operates as a "not" operator for the following characters in +replaceables+:
  *
- *  Both strings may use the <code>c1-c2</code> notation to denote ranges of
- *  characters, and +from_str+ may start with a <code>^</code>, which denotes
- *  all characters except those listed.
+ *      'hello'.tr('^aeiou', '-') # => "-e--o"
+ *      'hello'.tr('ae^iou', '-') # => "h-ll-"
```
Comment (peterzhu2118) [inline string.c:8053]: I'm a bit confused as to what this example is suggesting. Is it suggesting that a `^` is not treated as a special character unless it's the first character in the `replaceables` string? If that's the case, I think that it should be made more obvious. E.g.

```suggestion
 *      '^hello'.tr('ae^iou', '-') # => "-h-ll-"
```

File: string.c [hunk 2]
```diff
@@ -8022,37 +8022,50 @@ rb_str_tr_bang(VALUE str, VALUE src, VALUE repl)
...
+ *    operates as an escape character:
  *
- *     "hello\r\nworld".tr("\r", "")   #=> "hello\nworld"
- *     "hello\r\nworld".tr("\\r", "")  #=> "hello\r\nwold"
- *     "hello\r\nworld".tr("\\\r", "") #=> "hello\nworld"
+ *      'hel^lo'.tr('\^aeiou', '-')  # => "h-l-l-"    # Escaped leading circumflex.
+ *      'i-b-m'.tr('b\-z', 'a-z')    # => "ibabm"     # Escaped embedded hyphen.
+ *      'foo\\bar'.tr('ab\\', 'XYZ') # => "fooZYXr"   # Escaped backslash.
+ *      'abc\ndef'.tr('a-z', 'b-z')  # => "bcd\\oefg" # '\n' is two characters.
+ *      "abc\ndef".tr('a-z', 'b-z')  # => "bcd\nefg"  # "\n" is one newlne character.
```
Comment (peterzhu2118) [inline string.c:8067]: There's a typo

```suggestion
 *      "abc\ndef".tr('a-z', 'b-z')  # => "bcd\nefg"  # "\n" is one newline character.
```

Comment (peterzhu2118) [inline string.c:8067]: I'm not sure what these two examples are trying to show except that single quoted strings treat backslash characters literally whereas double quotes treat them as a control character.

Maybe the original examples showing how control characters can be replaced should be kept.

```
 *     "hello\r\nworld".tr("\r", "")   #=> "hello\nworld"
 *     "hello\r\nworld".tr("\\r", "")  #=> "hello\r\nwold"
 *     "hello\r\nworld".tr("\\\r", "") #=> "hello\nworld"
```

File: string.c [hunk 3]
```diff
@@ -8022,37 +8022,50 @@ rb_str_tr_bang(VALUE str, VALUE src, VALUE repl)
...
- *  Returns a copy of +str+ with the characters in +from_str+ replaced by the
- *  corresponding characters in +to_str+.  If +to_str+ is shorter than
- *  +from_str+, it is padded with its last character in order to maintain the
- *  correspondence.
+ *  Three characters may get special treatment:
  *
- *     "hello".tr('el', 'ip')      #=> "hippo"
- *     "hello".tr('aeiou', '*')    #=> "h*ll*"
- *     "hello".tr('aeiou', 'AA*')  #=> "hAll*"
+ *  - A circumflex (<tt>'^'</tt>) in +replaceables+
```
Comment (peterzhu2118) [inline string.c:8049]: Is this commonly called a circumflex? I've also seen it called a caret. I don't really know which one is correct 😅.

The documentation for `String#count` calls it a caret. We should decide on either circumflex or caret for consistency.

File: string.c (hunk 1 above)
Comment (BurdetteLamar) [reply string.c]: Fixed.

File: string.c (hunk 2 above)
Comment (BurdetteLamar) [reply string.c]: Fixed.

File: string.c (hunk 3 above)
Comment (BurdetteLamar) [reply string.c]: Caret is correct.  A circumflex is a diacritical mark; a caret is a character.

**Comment by BurdetteLamar** [comment]
@peterzhu2118, is this ready to merge?

**Pull request merged**